### DISCUSSION TITLE
Detail Documentation

### DISCUSSION THREAD

**Discussion by Fahad-Md-Kamal** [NilCoalescing/djangochannelsrestframework#55] [Q&A]
Hi, Hishnash.

I believe your project would be very much promising for the beginners like me if you had provided a little bit more detailed documentation.

Actually I'm trying to create a chat application with Django Channels which is very confusing. But yours seems would ease the burdain if there is a detail documentation. 

N.B: I'm doing my Final Year project with Django Rest Framework which is why may be your short Documentation is very confusing for me to apply in my project. 

Please, help me with any resources to implement your project in my solution.

**Comment by hishnash** [comment]
Hi @Fahad-Md-Kamal 

sorry about the documentation, this is still a very much work in progress part of the project.

> Actually I'm trying to create a chat application with Django Channels

For this application are you intending on saving the chat to the database?

**Comment by Fahad-Md-Kamal** [comment]
Yes, Hishnash... I want to save the chat messages to my database and retrieve those.

**Comment by zOthix** [comment]
Any update?

**Comment by lautarodapin** [comment]
I'm not in the computer, I will answer with more detail later.
You should use models for messages and rooms
Use actions for joining, saving room code and creating the message.
The  you should subscribe to messages in current room.
https://github.com/lautarodapin/noticias-vm/blob/main/chat/consumers.py check this that I made myself

**Reply by hishnash** [reply]
This is a really good example, do you mind if it take your example add some extra doc strings extra and use it when writing docs for this package?

**Reply by lautarodapin** [reply]
No problem, I was trying to writte the docstrings, but caught myself in something else.

**Comment by the-wc** [comment]
Any word on the update on this? The documentation on "Subscribing to a filtered list of models" is puzzling.

**Reply by hishnash** [reply]
sorry about this, I plan to try to re-hash this part of the docs soon I think I will attempt to create a visual diagram to explain things. Since the concept of how this works is very different from the regular Django request response flow its not an easy thing to explain with words (and my writing skills are rather sub-optimal).
